app.py

```diff
@@ -211,27 +212,30 @@
 
     # Set the column variable names.
     # This is meant to be flexible to accomodate a variable number of column names in the future
-    if "images/"+serial_number_string+".png" != "N/A":
-        if counter == 1:
-            column_names = list(data.keys())
-            for column_number, name in enumerate(column_names):
-                worksheet.write(0, column_number, name)
-            # add image column name
-            worksheet.write(0, len(list(data.keys()))+1, 'image')
-
-        # Set the actual data into the columns
-        trademark_data = list(data.values())
-        for column, trademark in enumerate(trademark_data):
-            worksheet.write(counter, column, trademark)
+
+    if counter == 1:
+        column_names = list(data.keys())
+        for column_number, name in enumerate(column_names):
+            worksheet.write(0, column_number, name)
+        # add image column name
+        worksheet.write(0, len(list(data.keys()))+1, 'image')
+
+    # Set the actual data into the columns
+    trademark_data = list(data.values())
+    for column, trademark in enumerate(trademark_data):
+        worksheet.write(counter, column, trademark)
+
+    # See if the path exists or not. If yes then look for img and add it, else nothing it will skip
+    if exists("images/" + serial_number_string + ".png"):
         worksheet.insert_image(counter, len(list(data.keys()))+1, 'images/'+serial_number_string+'.png', {'object_position': 1})
 
         # Set height of the row so its not awkward and squished
         img = Image.open('images/'+serial_number_string+'.png')
         height = img.height
         worksheet.set_row(counter, height)
 
-        # Increase counter by 1
-        counter += 1
+    # Increase counter by 1
+    counter += 1
```

Comment:
this should be outside the if exists block or else the images will become misordered with the rest of the data.